### PR TITLE
#2161 Add ignore-chiral-flag parameter to ketcher settings

### LINKED ISSUE #2161: Add ignore-chiral-flag parameter to ketcher settings
**Background**
Chiral flag is a legacy property from MDL molfiles. It should not be used

**Solution**
* Add a parameter ```Ignore the chiral flag``` inside Settings/Stereochemistry. Check box, default off. Location after "Text of Mixed flag"
* Add help text for mouse hover "Ignore chiral flag while loading from molfiles. By default all the stereo will be ABS"
* Add API parameter ```ignoreChiralFlag``` (Boolean; default=false) which can be changed during loading Ketcher editor
* Update help documentation for the flag behavior using this document
[ignore_chiral_flag.pdf](https://github.com/epam/ketcher/files/10743177/ignore_chiral_flag.pdf)

[ignore_chiral_flag.docx](https://github.com/epam/ketcher/files/10743178/ignore_chiral_flag.docx)

* if   ```ignoreChiralFlag``` = true 
  * input structures are treated as ABS until any enhanced stereo is defined.
  * do not display any labels ( "ABS" or "AND Enantiomer" etc) 
  * all enhanced stereo should be displayed always (related to #2160)
  * saving into molfile v2000 should set chiral flag = 1 (see doc file)

### PR DESCRIPTION
**Pull request by ansivgit**
Closes #2161 & #2160

### CODE REVIEW

File: packages/ketcher-core/src/application/ketcher.ts [hunk 1]
```diff
@@ -111,41 +115,36 @@ export class Ketcher {
...
+    )
 
-    if ('dearomatize-on-load' in options) {
-      return {
-        'general.dearomatize-on-load': options['dearomatize-on-load']
-      }
-    }
-    if ('ignoreChiralFlag' in options) {
-      return {
-        'general.ignore-no-chiral-flag': options.ignoreChiralFlag
```
Comment (KonstantinEpam23) [inline packages/ketcher-core/src/application/ketcher.ts]: @ansivgit I dont think this needs to be in server-style, I changed "general.ignore-no-chiral-flag" to "ignoreChiralFlag" here - do you think this is OK?

File: packages/ketcher-core/src/application/formatters/formatterFactory.ts [hunk 2]
```diff
@@ -56,6 +60,11 @@ export class FormatterFactory {
       molfileParseOptions.badHeaderRecover = badHeaderRecover
     }
 
+    if (typeof ignoreChiralFlag === 'boolean') {
+      molfileParseOptions.ignoreChiralFlag = ignoreChiralFlag
+      structServiceOptions['ignore-no-chiral-flag'] = ignoreChiralFlag
```
Comment (KonstantinEpam23) [inline packages/ketcher-core/src/application/formatters/formatterFactory.ts:65]: Just curios about the flag name "ignore-no-chiral-flag" - is it already supported by Indigo?

File: packages/ketcher-core/src/application/ketcher.ts (hunk 1 above)
Comment (ansivgit) [reply packages/ketcher-core/src/application/ketcher.ts]: Ok

File: packages/ketcher-core/src/application/formatters/formatterFactory.ts [hunk 3]
```diff
@@ -70,6 +79,7 @@ export class FormatterFactory {
     switch (format) {
       case SupportedFormat.ket:
         formatter = new KetFormatter(new KetSerializer())
+        // formatter = new KetFormatter(new KetSerializer(molSerializerOptions))
```
Comment (Konstantin1996) [inline packages/ketcher-core/src/application/formatters/formatterFactory.ts:82]: Do you need it? Please, remove

File: packages/ketcher-core/src/domain/serializers/mol/molSerializer.ts [hunk 4]
```diff
@@ -37,27 +37,34 @@ export class MolSerializer implements Serializer<Struct> {
...
+          })
         } catch (ex1) {
           //
         }
         try {
           // check for a missing first line
           // this sometimes happens when pasting
-          return molfile.parseCTFile(
-            [''].concat(lines),
-            this.options.reactionRelayout
```
Comment (Konstantin1996) [inline packages/ketcher-core/src/domain/serializers/mol/molSerializer.ts:59]: this.options.reactionRelayout is not used anymore in molSerializer.ts.

I think we can delete it from class itself and from other methods like:

parseCTFile, parseRxn...

**Review by KonstantinEpam23 — APPROVED** [review]

**Pull request merged**